core/src/main/java/org/infinispan/transaction/tm/DummyTransaction.java

```diff
@@ -45,7 +47,7 @@
    private final Xid xid;
    private volatile int status = Status.STATUS_UNKNOWN;
    private final List<Synchronization> syncs;
-   private final List<XAResource> resources;
+   private final Map<XAResource, Integer> resources;
```

Comment:
I would prefer if you created another ```List``` with the prepared resources. 
I'm not sure if the method ```isSameRM()``` and ```equals()``` has to be equivalent and it can happen the ```Map``` to replace a ```XaResource``` because of ```equals()``` returns ```true```.